### ISSUE TITLE
CLIEngine API to resolve config name to absolute path

### ISSUE THREAD

**Issue by sindresorhus** [eslint/eslint#7328]
Edit to clarify:

In [XO](https://github.com/sindresorhus/xo), I accept similar options like ESLint does in `package.json`. One of these are `extends`. When a user specifies `"xo": {"extends": "react"}` in package.json, XO reads it, adds it to `baseConfig.extends` and passes it to ESLint. The problem is that the user expects `react` to be resolved from their package.json's location, but `baseConfig.extends` is resolved from `__dirname`, which is the folder of XO, not the user's project. I've worked around this until now by manually resolving to an absolute path first, but my resolution logic is buggy. I would like to use the same logic ESLint uses internally to resolve the path of a config with a custom `cwd`.

---

The `baseConfig` option is very useful, but it seems to resolve using `__dirname` and not the `cwd` option for the `extends` config property. This problematic as I need to take a user-supplied array of configs to extend and pass them to `baseConfig.extends`, but they wouldn't be resolved correctly, as I need to resolve from `process.cwd()`, not `__dirname`. So I first need to resolve them to absolute paths. I currently do that using a [naive and buggy implementation](https://github.com/sindresorhus/xo/blob/3c428474cd51579ed55ac94f8e9a467cd48fe9ac/options-manager.js#L127-L152). It would be very helpful if I could use the same machinery as ESLint uses internally.

---

I'm thinking:

``` js
eslint.CLIEngine.resolveConfigPath(name, [cwd]);
```

With `cwd` being `process.cwd()` by default.

---

ESLint 3.7.1

**Comment by platinumazure** [comment]
Thanks @sindresorhus for your proposal. Seems reasonable to me, but being a core/CLI question, it needs to be discussed by our Technical Steering Committee. If this issue is not accepted before the next TSC meeting, it will be discussed there.

TSC Summary: See proposal in first post (will summarize later)

TSC Question: Shall we accept this proposal and put this issue on the Core Roadmap?

**Comment by nzakas** [comment]
@sindresorhus can you please rephrase your request by describing in more detail what your use case is? 

I can't really evaluate the proposed solution without better understanding the problem you're running into.

**Comment by eslintbot** [comment]
Hi @sindresorhus, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by platinumazure** [comment]
At @nzakas' request I am changing the labels to reflect that we need more information.

@sindresorhus I *think* I understand what you're trying to do, but just to be safe, can you walk us through the actual problem you're trying to solve? You mentioned processing user inputs that are config paths-- could you tell us some more about that? Thanks!

**Comment by vitorbal** [comment]
friendly ping @sindresorhus, could you check @platinumazure's questions above when you have a chance?

**Comment by sindresorhus** [comment]
Sorry about the delay. I've edited the original issue text to make it clearer. Let me know if there's anything else I need to clarify.

**Comment by nzakas** [comment]
Thanks. I think this might actually be a bug in the core, but need some time to investigate and confirm.

**Comment by kaicataldo** [comment]
I'll see if I can find some time to look into this. If anyone else can get to it first, please feel free!

**Comment by not-an-aardvark** [comment]
Is this still an issue? The config internals have been refactored a lot since this was originally reported.

**Comment by not-an-aardvark** [comment]
Closing because there hasn't been any activity here for awhile and it's not clear if this is still a problem. Feel free to open a new issue if there is still a bug.

**Comment by SavePointSam** [comment]
I can confirm this is no longer an issue.
